[Qin Guorui] iP

qinguorui2001 opened the pull request:
### Guorui's Duke 
>*“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))*

Guorui's Duke frees your mind of having to remember things you need to do. It's,

- text-based

- easy to learn

- ~~FAST~~ SUPER FAST to use

All you need to do is,

  1.download it from [here](https://github.com/qinguorui2001/ip/releases/tag/A-Jar)

  2.open terminal.

  3.use ` cd ` command to navigate to the directory of jar file.

  4.enter ` java - jar individual project.jar `(Prerequisites: [jdk](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html) installed)    

  5.add your tasks.

  6.let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing Todo, Event and Deadline tasks

- [x]  Search for the tasks on specific day

- [x]  Find tasks with similar content

If you Java programmer, you can use it to practice Java too. Here's the main method:

For GUI:
```java
    public class Launcher {
        public static void main(String[] args) {
            Application.launch(Main.class, args);
        }
    }
```

Commit:
Add Gradle support

qinguorui2001 committed:
Level 1. Greet, Echo, Exit

qinguorui2001 committed:
Level 1, Change output format and add annotations

qinguorui2001 committed:
Level 2, achieve add and list functions

qinguorui2001 committed:
Level 3. Formulate the Task class.

qinguorui2001 committed:
Level 3.Allows the task status change operations and new print format

qinguorui2001 committed:
Level 4. Add sub tasks of Task, add toString method to Task, allow corresponding operations

qinguorui2001 committed:
A-TextUiTesting

qinguorui2001 committed:
Level 5. Add features of handling some exceptions.

qinguorui2001 committed:
Level 6. Replace some substring methods with startWith method, and modify task array to its array list form

qinguorui2001 committed:
Level 6.Delete operations

qinguorui2001 committed:
Tidy up code and update test cases

qinguorui2001 committed:
Level 6.Modify some throws annotation and changes the test file

qinguorui2001 committed:
Level 6.Correct test file

qinguorui2001 committed:
Update documentations of each class.

qinguorui2001 committed:
A-Enums,Super upgrade of the code and add operation enum

qinguorui2001 committed:
Super simplified verson

qinguorui2001 committed:
Fix bugs of done and delete print message of outofrange exception

qinguorui2001 committed:
Set up data file

qinguorui2001 committed:
Level 7.

qinguorui2001 committed:
Level 8.Add the funtions of matching date and certify date.

qinguorui2001 committed:
Merge branch 'branch-Level-7'

qinguorui2001 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

qinguorui2001 committed:
Resolves the conflict.

qinguorui2001 committed:
A-MoreOOP. Makes the structure more OOP.

qinguorui2001 committed:
A-Packages.Seperate using packages

qinguorui2001 committed:
A-Junit.Add tests for parser storage and command

qinguorui2001 committed:
A-Jar. Makes it runnable

qinguorui2001 committed:
A-JavaDoc.Add correct documentations.

qinguorui2001 committed:
A-JavaDoc.Coding standard Modifications.

qinguorui2001 committed:
Level 9.Find.

qinguorui2001 committed:
Merge branch 'branch-Level-9'

qinguorui2001 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

qinguorui2001 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

qinguorui2001 committed:
Add some features and documentations.

qinguorui2001 committed:
Debug the load method in Storage class, and Rebuild the jar

qinguorui2001 committed:
Update data file

qinguorui2001 committed:
Merge branch 'branch-A-CodingStandard'

qinguorui2001 committed:
Merge branch 'add-gradle-support' of https://github.com/qinguorui2001/ip

qinguorui2001 committed:
change gradle

qinguorui2001 committed:
checkstyle

qinguorui2001 committed:
Update using checkstyle.

Th-429B commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,70 @@
+package duke.command;
+
+import duke.storage.Storage;
+//
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+//
+import duke.ui.Ui;
```
I think we should standardize whether we separate package imports or not.

qinguorui2001 replied:
yes, i then deleted "//", and combined them

Th-429B commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("*****")
+                Command c = Parser.parse(fullCommand);
+                assert c != null;
```
I like the use of assert here!

qinguorui2001 replied:
thx

Th-429B commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,115 @@
+     * @param input String input.
+     * @return Content of input is digit or not.
+     */
+    public static boolean chekDigit(String input) {
```
should checkDigit be spelled like that?

qinguorui2001 replied:
oops, i haven't detected it before, nice catch!

Th-429B commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,70 @@
+     * @param storage The instance to store data.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
I think it would be good to specify what type corresponds to what action in the javadocs so that it will be easier to understand.

Th-429B reviewed (commented):
Overall, the coding standard is up to par, with only very minor improvements. Well done!

Th-429B commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,70 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task task = new Task("");
+        switch (type) {
+        case 1:
```
I think it would be good to specify what type corresponds to what action in the javadocs so that it will be easier to understand.

qinguorui2001 replied:
yes, i will change it using enum

qinguorui2001 committed:
Set gradle

qinguorui2001 committed:
level 10.

qinguorui2001 committed:
Merge branch 'branch-Level-10'

jhlee1997 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+    @FXML
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
```
Perhaps it is better to name this as ```pictureDisplay``` as this is a variable name.
```suggestion
    private ImageView pictureDisplay;
```

jhlee1997 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,39 @@
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws OutOfRangeException {
+        //check with the special response "delete X", where X is index of deleted item.
+        int curr = Integer.parseInt(response.substring(7));
```
Perhaps it is better to use a named constant like ```LEN_OF_DELETE``` to avoid magic numbers.
```suggestion
        int curr = Integer.parseInt(response.substring(LEN_OF_DELETE));
```

jhlee1997 commented on src/main/java/duke/exception/NotRecognizeException.java:
```diff
@@ -0,0 +1,12 @@
+ *
+ * @author QIN GUORUI
+ */
+public class NotRecognizeException extends DukeException {
```
Perhaps it is better to name this class as ```InvalidCommandException``` for better readability.
```suggestion
public class InvalidCommandException extends DukeException {
```

jhlee1997 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,217 @@
+        fw.close();
+    }
+
+    /**
+     * Stores the task to data file.
+     *
+     * @param task The string format of task representation.
+     */
```
Perhaps it is better to add a ```@throw``` for the JavaDoc of this method

jhlee1997 reviewed (commented):
Hi Qin Guorui, I am Jae and this is my second PR review about code quality.
Overall, I really appreciate the efforts you have put in to make a nice Duke application.
Just picked out 4 minor adjustments to make. Thanks!

qinguorui2001 committed:
More features related to GUI

qinguorui2001 committed:
Deal with bye in GUI

qinguorui2001 committed:
UI related classes: update GUI to make it beutiful

The Ui classes are not well defined and the GUI looks terrible.

Let's modify the MainWindow and DialogBox classes to add more
features related to background color of panel and text field, and
change the font and color of texts.

This ensures the good looking of GUI and the texts are more
readable towards users.

qinguorui2001 committed:
Add some assertions related to invariants in the methods.

qinguorui2001 committed:
All classes: improve code quality.

These classes include arrow head phenomenon, complex expressions
and long methods, which are not permitted.

Let's refactor common parts of methods and divide them into smaller
methods, and rename some functions to make them not misleading.

Using techniques taught in class could largely improve the code
quality, they are simple to apply and maintain the code.

qinguorui2001 committed:
A-Streams

qinguorui2001 committed:
Achieve B-Reminders

qinguorui2001 committed:
Merge pull request #1 from qinguorui2001/branch-B-Reminders

Achieve B-Reminders

qinguorui2001 committed:
Merge branch 'master' into branch-A-Assertions

qinguorui2001 committed:
Merge pull request #2 from qinguorui2001/branch-A-Assertions

Add some assertions related to invariants in the methods.

qinguorui2001 committed:
Merge branch 'master' into branch-A-Streams

qinguorui2001 committed:
Merge pull request #3 from qinguorui2001/branch-A-Streams

A-Streams

qinguorui2001 committed:
Merge branch 'master' into branch-A-CodeQuality

qinguorui2001 committed:
Merge pull request #4 from qinguorui2001/branch-A-CodeQuality

All classes: improve code quality.

qinguorui2001 committed:
Update BCD extension B-reminders

qinguorui2001 committed:
Further improve the code quality

qinguorui2001 committed:
Update the workflow file

qinguorui2001 committed:
A better gui

qinguorui2001 committed:
Ui.png

qinguorui2001 committed:
Update README.md

qinguorui2001 committed:
Adapt zhang shichen's wonderful idea

qinguorui2001 committed:
Merge branch 'master' of https://github.com/qinguorui2001/ip

qinguorui2001 committed:
modify small bug of image repeat

qinguorui2001 committed:
final version

qinguorui2001 committed:
final improvement in BCD extension

qinguorui2001 committed:
Create gradle.yml

qinguorui2001 committed:
anyway further improvement in javadoc

qinguorui2001 committed:
Merge branch 'master' of https://github.com/qinguorui2001/ip

qinguorui2001 committed:
Update README.md
